pkgs/by-name/ba/basalt-monado/package.nix

```
@@ -0,0 +1,110 @@
+{ stdenv,
+  lib,
+  fetchFromGitLab,
+  fetchFromGitHub,
+  autoPatchelfHook,
+  boost,
+  bzip2,
+  cereal,
+  cli11,
+  cmake,
+  eigen,
+  extra-cmake-modules,
+  fmt,
+  freeglut,
+  glew,
+  libGL,
+  magic-enum,
+  opencv,
+  opengv,
+  pangolin,
+  pkg-config,
+  sophus,
+  tbb,
+  nix-update-script
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "basalt-monado";
+  version = "release-673cc5c6";
```

```suggestion
  version = "0-release-673cc5c6";
```
Versions must start with a number: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md?plain=1#L350

---

That's the release tag upstream uses https://gitlab.freedesktop.org/mateosss/basalt/-/tags

but we would strip the release from it.

---

This is not really a meaningful version IMO.

---

why is this not being fixed? this is a nonsensical version.

---

Because:

a) Versions must start with a digit
b) This is not an "unstable" version; this is actually a release
c) The short commit identifier of the release is `673cc5c6`.

---

> a) Versions must start with a digit

Which this clearly violates.


> c) The name of the git tag is `release-673cc5c6`.

And again, this is not meaningful, it's just some rando git commit, that's not a numerical version.

---

> Which this clearly violates.

Oh, I thought they'd applied my patch. Yea, that needs fixing.

> And again, this is not meaningful, it's just some rando git commit, that's not a numerical version.

There are no numerical versions to basalt-monado. According to the applicable [naming guidelines](https://github.com/NixOS/nixpkgs/blob/ff93fa8385374ab39eafc2dc882f699bc010f49a/pkgs/README.md?plain=1#L344-L362)

- The `pname` and the `version` attribute _must not_ contain uppercase letters — e.g., `"mplayer"` instead of `"MPlayer"`.

Satisfied.

- The `version` attribute _must_ start with a digit e.g., `"0.3.1rc2"` or `"0-unstable-1970-01-01"`.

Not satisfied, ergo my patch.

 - If a package is a commit from a repository without a version assigned, then the `version` attribute _should_ be the latest upstream version preceding that commit, followed by `-unstable-` and the date of the (fetched) commit. The date _must_ be in `"YYYY-MM-DD"` format.

Not applicable, as there is an assigned version.

- If a project has no suitable preceding releases - e.g., no versions at all, or an incompatible versioning / tagging schema - then the latest upstream version in the above schema should be `0`.

Not applicable, as there are releases.

So, in summary: the only blocking change is to prefix the release with a digit.

---

Sorry for taking so long to say decide on how to handle this, I was waiting for what the consensus would be for it and only now had time to properly see this, I decided to go with this part of the guideline:
> If a project has no suitable preceding releases - e.g., no versions at all, or an incompatible versioning / tagging schema - then the latest upstream version in the above schema should be `0`.
> 
> Example: Given a project that has no tags / released versions at all, or applies versionless tags like `latest` or `YYYY-MM-DD-Build`, a commit authored on March 15, 2022 would have `version = "0-unstable-2022-03-15"`. 

So the version is now `0-unstable-2024-06-21`.

---

I also upgraded to the latest "release" on the repo.